needle.pm

```diff
@@ -195,7 +202,9 @@ sub wanted_ {
 }
 
 sub init {
-    $needledir = shift if @_;
+    $needledir    = shift if @_;
+    $shared_cache = shift if @_;
```

Comment:
What do you mean by 'in one cache'? I expect init sub to change everything needed. IOW I don't expect init subs to be idempotent.